src/d3d10/d3d10_main.cpp

```diff
@@ -41,6 +41,10 @@ extern "C" {
     hr = D3D11CoreCreateDevice(pFactory, pAdapter,
       Flags, &FeatureLevel, 1, &d3d11Device);
     
+    Com<ID3D10Multithread> multithread;
+    d3d11Device->QueryInterface(__uuidof(ID3D10Multithread), reinterpret_cast<void**>(&multithread));
+    multithread->SetMultithreadProtected(!(Flags & D3D10_CREATE_DEVICE_SINGLETHREADED));
+
     if (FAILED(hr))
       return hr;
```

Comment:
The `if (FAILED(hr))` check needs to be moved up before setting up the MT protection, otherwise things will crash if device creation fails (and it can legitimately fail).